common/reviews/api/rush-lib.api.md

```diff
@@ -678,9 +744,13 @@ export class _RushGlobalFolder {
     get path(): string;
 }
 
-// @beta (undocumented)
+// @beta
 export class RushLifecycleHooks {
-    initialize: AsyncSeriesHook<void>;
+    anyGlobalScriptCommand: AsyncSeriesHook<IGlobalScriptAction>;
+    anyPhasedScriptComamnd: AsyncSeriesHook<IPhasedScriptAction>;
+    globalScriptCommand: HookMap<AsyncSeriesHook<IGlobalScriptAction>>;
+    initialize: AsyncSeriesHook<IRushAction>;
+    phasedScriptCommand: HookMap<AsyncSeriesHook<IPhasedScriptAction>>;
```

Comment:
These hook names don't really describe when they're called? Are these called when these things are constructed? Invoked? What's the difference between `anyGlobalScriptCommand` and `globalScriptCommand`?

Hook names should probably include a verb that describes when they're invoked.

Reply:
I would love for them to be invoked the moment the parser cares about them, such that you could, for example, modify the CLI parameters. Unfortunately, `ts-command-line` doesn't support deferring the parser specification for an action until it is the action being parsed (or `--help` was requested, in which case effectively all actions are the action being parsed).

I can prefix these all with `run` if you want to make them exact.

If you have a global script command `rush foo`, the following execute in order when you invoke it:
```
// Run the general hook for any Rush action
await hooks.initialize.promise(fooAction);
// Run the general hook for a global script action
await hooks.anyGlobalScriptCommand.promise(fooAction);
// Run the specific hook for a global script action named "foo"
await hooks.globalScriptCommand.get('foo')?.promise(fooAction);
```

The purpose of the increasing levels of specificity is so that the lazy-install/lazy-load of Rush plugins can be scoped to:
- Any Rush action
- Any global script action
- A global script action named "foo"
- etc.

Reply:
By this convention, `initialize` is really `anyCommand`